### PR TITLE
DASH-1168 get wrap and burn domain actions refactor

### PR DESCRIPTION
**Pull request by trukhilio**
- Refactor Burn domains flow
- Refactor wrap tokens/domains flow
- Code enhancements and bug fixes

### CODE REVIEW

File: controller/api/fio.js [hunk 1]
```diff
@@ -1121,22 +818,21 @@ class FIOCtrl {
...
           }
         }
-      };
+      } else {
+        console.log(`${logPrefix}No domains to burn.`);
+      }
 
-      await processActions();
+      console.log(
+        `${logPrefix}Update FIO Block Number for burn NFTS: ${lastIrreversibleBlock}`,
```
Comment (andreyvEze) [inline controller/api/fio.js:826]: do we need backspace after prefix like in other places?

File: controller/utils/fio-chain.js [hunk 2]
```diff
@@ -1,27 +1,101 @@
...
+        : null,
+    })
+  ).json();
+
+const getFioBlockInfo = async (lastIrreversibleBlock) =>
+  await (
+    await fetchWithRateLimit({
+      url: makeGetBlockUrl(FIO_SERVER_URL_ACTION),
+      options: {
+        body: `{"block_num_or_id": ${lastIrreversibleBlock}}`,
```
Comment (andreyvEze) [inline controller/utils/fio-chain.js:57]: would it better to use `JSON.stringify({block_num_or_id: lastIrreversibleBlock})`

File: controller/utils/fio-chain.js [hunk 3]
```diff
@@ -37,146 +111,193 @@ export const getLastIrreversibleBlockOnFioChain = async () => {
...
+      reverse: true,
+    };
 
-  return actionsHistory;
+    const response = await getTableRows({
+      logPrefix,
+      tableRowsParams,
+    });
+
+    return response && response.rows[0] ? response.rows[0].id + 1 : null; // Set id + 1 because if we will set id as it is then it will process already processed wrap item
```
Comment (andreyvEze) [inline controller/utils/fio-chain.js:153]: does it always exist? `response.rows[0].id + 1`

Comment (andreyvEze) [reply controller/utils/fio-chain.js:153]: and should weforce number type for `response.rows[0].id`

**Review by andreyvEze — APPROVED** [review]

**Review by nicolasMazuryk — APPROVED** [review]

File: controller/api/fio.js (hunk 1 above)
Comment (trukhilio) [reply controller/api/fio.js:826]: Done

File: controller/utils/fio-chain.js (hunk 2 above)
Comment (trukhilio) [reply controller/utils/fio-chain.js:57]: Done

File: controller/utils/fio-chain.js (hunk 3 above)
Comment (trukhilio) [reply controller/utils/fio-chain.js]: Done

**Pull request merged**